guides/source/active_record_migrations.md

```diff
@@ -931,43 +1183,66 @@ number associated with the previous migration you can run:
 $ bin/rails db:rollback
 ```
 
-This will rollback the latest migration, either by reverting the `change`
-method or by running the `down` method. If you need to undo
-several migrations you can provide a `STEP` parameter:
+This will rollback the latest migration, either by reverting the `change` method
+or by running the `down` method. If you need to undo several migrations you can
+provide a `STEP` parameter:
 
 ```bash
 $ bin/rails db:rollback STEP=3
 ```
 
 The last 3 migrations will be reverted.
 
-The `db:migrate:redo` command is a shortcut for doing a rollback and then migrating
-back up again. As with the `db:rollback` command, you can use the `STEP` parameter
-if you need to go more than one version back, for example:
+In some cases where you modify a local migration and would like to rollback that
+specific migration before migrating back up again, you can use the
+`db:migrate:redo` command. As with the `db:rollback` command, you can use the
+`STEP` parameter if you need to go more than one version back, for example:
 
 ```bash
 $ bin/rails db:migrate:redo STEP=3
 ```
 
-Neither of these rails commands do anything you could not do with `db:migrate`. They
-are there for convenience, since you do not need to explicitly specify the
-version to migrate to.
+NOTE: You could get the same result using `db:migrate`. However, these are there
+for convenience so that you do not need to explicitly specify the version to
+migrate to.
 
-### Setup the Database
+#### Transactions
 
-The `bin/rails db:setup` command will create the database, load the schema, and initialize
-it with the seed data.
+In databases that support transactions with statements that change the schema,
+each migration is wrapped in a transaction.
+
+INFO: A transaction ensures that if a migration fails partway through, any
+changes that were successfully applied are rolled back, maintaining database
+consistency. This means that either all operations within the transaction are
+executed successfully, or none of them are, preventing the database from being
+left in an inconsistent state if an error occurs during the transaction.
+
+If the database does not support transactions with statements that change the
+schema, then when a migration fails, the parts of it that have succeeded will
+not be rolled back. You will have to rollback the changes manually.
+
+NOTE: There are certain queries that can't run inside a transaction. If your
+adapter supports DDL transactions you can use `disable_ddl_transaction!` method
+to disable transactions for a single migration, allowing these queries to be
+executed outside the transactional boundary.
```

Comment:
Maybe we can replace this NOTE with the [example from the API docs](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Migration.html#class-ActiveRecord::Migration-label-Transactional+Migrations), and avoid the term "the transactional boundary".
````suggestion
There are queries that you can’t execute inside a transaction though, and for these situations you can turn the automatic transactions off with `disable_ddl_transaction!`:

```ruby
class ChangeEnum < ActiveRecord::Migration[8.0]
  disable_ddl_transaction!

  def up
    execute "ALTER TYPE model_size ADD VALUE 'new_value'"
  end
end
```
NOTE: Remember that you can still open your own transactions, even if you are in a Migration with self.disable_ddl_transaction!.
````

Reply:
Love this, thank you!